samples/99_zil_interpreter/tests/parser_tests_basic.rb

```diff
@@ -0,0 +1,355 @@
+# test basic SETG command with STRING and FIX
+# -----------------------------------------
+def test_parser_01(args, assert)
+  source = <<-ZIL
+<SETG ZORK-NUMBER 1>
+  ZIL
+
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :SETG,
+    :"ZORK-NUMBER",
+    1)
+
+  assert.equal! parsed, expected
+end
+
+# test escaped atom
+# -----------------------------------------
+def test_parser_02(args, assert)
+  # need to escape backslash in HEREDOC to get correct result
+  source = <<-ZIL
+<STRING !\\" !,WBREAKS>
+  ZIL
+
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :STRING,
+    :"!\"",
```

Comment:
According to "7.6.6. STRING (the PRIMTYPE) and CHARACTER [1]" in

https://mdl-language.readthedocs.io/en/latest/07-structured-objects/

The ZIL code
```zil
!\"
```
is representing the single character `"` I think.

So this line should probably be 
```rb
'"',
```

Reply:
The original statement in zork1.zil is this:

```
<OR <GASSIGNED? ZILCH>
    <SETG WBREAKS <STRING !\" !,WBREAKS>>>
```

I had to escape the HEREDOC so it would preserve the \ character.

Since this is an ATOM, and not a STRING, I believe it is governed by the rules in "2.6.3.3. \ (Backslash) in ATOMs", where the \ character is saying the atom's name is `!"`.

Reply:
I'm quite sure it is a character...... 

1. The builtin function `STRING` only accepts strings or characters as arguments according to

  https://mdl-language.readthedocs.io/en/latest/appendix-2-predefined-subroutines/
  
  (Search for `STRING ("TUPLE" ELEMENTS)` - the DECL statement will tell you so)
  
2. If you look at "7.7.3. Examples [1]" in https://mdl-language.readthedocs.io/en/latest/07-structured-objects/

  you will see a list of characters
  ```zil
  (!\S !\T !\R !\U !\N !\G !\.)
  ```

Actually there is also another syntax construct here: `!,WBREAKS` the exclamation mark here is a Segment 
That is basically a splat operator in modern terms - it turns the list-like content of it's argument into single elements and passes them directly as elements....... in this case.... taking the single characters of the string contained in `WBREAKS` and passes them as single arguments to the STRING function.......


I think translated roughly to ruby this line would look something like this:
```rb
if $ZILCH
  $WBREAKS = ['"', *$WBREAKS.chars].join
end
```

Reply:
You are correct.  

I found the character definition in "7.6.6. STRING (the PRIMTYPE) and CHARACTER [1]".  (`!\any-ASCII-character`)

Reply:
Segments and characters will be implemented in next commit.  It looks the WBREAKS variable is something that was used by whatever called zork1.zil.  I don't think our ZIL interpreter will actually do anything with this instruction. (used by ZAP compiler?)